src/sources/ldap.rs

```diff
@@ -124,13 +124,11 @@ impl LdapSource {
 fn read_string_entry(
 	entry: &SearchEntry,
 	attribute: &AttributeMapping,
-	id: &str,
+	_id: &str,
 ) -> Result<String> {
 	match read_search_entry(entry, attribute)? {
 		StringOrBytes::String(entry) => Ok(entry),
-		StringOrBytes::Bytes(_) => {
-			Err(anyhow!("Unacceptable binary value for {} of user `{}`", attribute, id))
-		}
+		StringOrBytes::Bytes(bytes) => Ok(BASE64_STANDARD.encode(&bytes)),
```

Comment:
@tlater-famedly
I couldn't immediately understand why wasn't Bytes accepted?
Also, shall Base64 be sufficient here?

Reply:
This function's entire point was throwing an error for non-string values, it's only here to not write the same error handling 6 times x) Hence the doc comment:

```rust
/// Read an an attribute, but assert that it is a string
```

I don't think base64 encoding these values is a good idea, it will be stuff like user names, which should always be strings. We probably should skip syncing users which have non-string-encoded values that we expect to be strings.

This is different for the values that we expect to be raw binary, for those we use the `read_search_entry` function, and handle the non-string case explicitly.

This *is* indeed a change, but I think this just makes more sense. Happy to hear counter arguments.

Reply:
Okay, reverted.